### PR TITLE
Add new loss function based on profit/drawdown ratio per pair

### PR DESCRIPTION
**Pull request by mrpabloyeah**
## Summary

A new loss function with a different approach than those currently available

### CODE REVIEW

**Review by xmatthias — COMMENTED** [review]
Wouldn't optimizing for a good median / avg make more sense? this way you can have one negative outlier mess with your overall result.

That said - i'd apreciate if you could
* add it to the `test_loss_functions_better_profits` test (so it's at least running once - it doesn't test logic too much)
* add it to `HYPEROPT_LOSS_BUILTIN` in constants (so it's showing up in help) 
* add a quick explanation/description to the hyperopt.md file (so people know it exists).

make sure to run `build_helpers/create_command_partials.py` afterwards - and commit that result - otherwise CI will tell you to :laughing:

**Comment by mrpabloyeah** [comment]
> Wouldn't optimizing for a good median / avg make more sense? this way you can have one negative outlier mess with your overall result.

It makes sense, but this approach seeks robustness rather than average profitability. Optimizing for the worst case prevents a pair with very poor results from being hidden by the average. It's a way to minimize risk and prevent Hyperopt from overfitting only certain strong pairs.

File: docs/hyperopt.md [hunk 1]
```diff
@@ -471,6 +471,7 @@ Currently, the following loss functions are builtin:
 * `SortinoHyperOptLossDaily` - optimizes Sortino Ratio calculated on **daily** trade returns relative to **downside** standard deviation.
 * `MaxDrawDownHyperOptLoss` - Optimizes Maximum absolute drawdown.
 * `MaxDrawDownRelativeHyperOptLoss` -  Optimizes both maximum absolute drawdown while also adjusting for maximum relative drawdown.
+* `MaxDrawDownPerPairHyperOptLoss` -  Optimizes the profit/drawdown ratio calculated individually for each pair in the pairlist.
```
Comment (xmatthias) [inline docs/hyperopt.md]: I think this doesn't really explain the point of this well.
the docstring of the function itself is better, really

Comment (mrpabloyeah) [reply docs/hyperopt.md]: > I think this doesn't really explain the point of this well. the docstring of the function itself is better, really

You're right! It's already fixed, now it's much more descriptive.

**Review by xmatthias — APPROVED** [review]
great addition :+1:
